odoo/cli/command.py

```diff
@@ -17,54 +20,79 @@ class Command:
 
     def __init_subclass__(cls):
         cls.name = cls.name or cls.__name__.lower()
+        module = cls.__module__.rpartition('.')[2]
+        if not cls.is_valid_name(cls.name):
+            raise ValueError(
+                f"Command name '{cls.name!r}' "
+                f"must match '{COMMAND_NAME_RE.pattern}'")
+        if cls.name != module:
+            raise ValueError(
+                f"Command name '{cls.name!r}' "
+                f"must match Module name '{module!r}'")
         commands[cls.name] = cls
 
+    @classmethod
+    def is_valid_name(cls, name):
+        return re.match(COMMAND_NAME_RE, name)
+
 
 def load_internal_commands():
-    """Load `commands` from `odoo.cli`"""
+    """Load ``commands`` from ``odoo.cli``"""
     for path in odoo.cli.__path__:
         for module in Path(path).iterdir():
             if module.suffix != '.py':
                 continue
             __import__(f'odoo.cli.{module.stem}')
 
 
-def load_addons_commands():
-    """Load `commands` from `odoo.addons.*.cli`"""
-    logging.disable(logging.CRITICAL)
-    initialize_sys_path()
-    for module in get_modules():
-        if (Path(get_module_path(module)) / 'cli').is_dir():
-            with contextlib.suppress(ImportError):
-                __import__(f'odoo.addons.{module}')
-    logging.disable(logging.NOTSET)
-    return list(commands)
+def load_addons_commands(paths, search_term="*"):
```

Comment:
```suggestion
def load_addons_commands(paths, command=None):
    if command is None:
        command = '*'
    elif not Command.is_valid_command(command):
        raise ValueError(command)
```
At the moment you can call this function like `load_addons_command(..., "../../../../../../../../path/to/malicious")` and load a malicious script. We should not allow that.